prefix.go

```diff
@@ -318,6 +318,10 @@ func (i *ipamer) acquireSpecificIPInternal(prefixCidr, specificIP string) (*IP,
 		if !ipnet.Contains(specificIPnet) {
 			return nil, fmt.Errorf("given ip:%s is not in %s", specificIP, prefixCidr)
 		}
+		_, ok := prefix.ips[specificIPnet.String()]
+		if ok {
+			return nil, fmt.Errorf("given ip:%s is already allocated", specificIP)
```

Comment:
Cool stuff!
Would you mind wrapping something like `ErrAlreadyAllocated` here,
so I can `errors.Is(err, goipam.ErrAlreadyAllocated)` when I use this function?

Obviously not part of this PR, but if I could wish for something :)
It would be awesome if callers could distinguish between 3 error states:
- transient errors (due to optimistic locking?)
- non transient/validation errors (stuff that will never work, like trying to allocate an IP not in the CIDR)
- already allocated IPs (So I can either retry later in the hope it is free or actually wait for a lease of the specific IP to expire)

Caller code relying on `goipam.ErrNoIPAvailable` being returned in this case:
https://github.com/routerd/kube-ipam/blob/separate-ipfamilies/internal/controllers/iplease_controller.go#L148

Reply:
good point, done for this case.